controller/node_controller.go

```diff
@@ -967,3 +951,84 @@ func (nc *NodeController) cleanUpBackingImagesInDisks(node *longhorn.Node) error
 
 	return nil
 }
+
+func BackingImageDiskFileCleanup(node *longhorn.Node, bi *longhorn.BackingImage, bids *longhorn.BackingImageDataSource, waitInterval time.Duration, haRequirement int) {
```

Comment:
This is a good step forward with the addition of the unit tests.
In general I would have liked to see a split between evaluation/decision making (which to delete) & modification (operates on the prior set of decisions) then removes them from the resources.

This way Decision logic can be independently tested, but this is good enough :)
Thank you for adding it.